src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteLeaderIT.kt

```diff
@@ -0,0 +1,81 @@
+package com.amazon.elasticsearch.replication.integ.rest
+
+import com.amazon.elasticsearch.replication.MultiClusterRestTestCase
+import com.amazon.elasticsearch.replication.MultiClusterAnnotations
+import com.amazon.elasticsearch.replication.StartReplicationRequest
+import com.amazon.elasticsearch.replication.startReplication
+import com.amazon.elasticsearch.replication.stopReplication
+import org.apache.http.entity.ContentType
+import org.apache.http.nio.entity.NStringEntity
+import org.assertj.core.api.Assertions
+import org.elasticsearch.client.Request
+import org.elasticsearch.client.RequestOptions
+import org.elasticsearch.client.indices.CreateIndexRequest
+import org.elasticsearch.client.indices.GetIndexRequest
+import org.elasticsearch.test.ESTestCase.assertBusy
+import org.junit.Assert
+import org.junit.Ignore
+import java.util.concurrent.TimeUnit
+
+@MultiClusterAnnotations.ClusterConfigurations(
+        MultiClusterAnnotations.ClusterConfiguration(clusterName = LEADER),
+        MultiClusterAnnotations.ClusterConfiguration(clusterName = FOLLOWER)
+)
+
+class ClusterRerouteLeaderIT : MultiClusterRestTestCase() {
```

Comment:
Nit - seems like there is potential to share code (esp changeTemplate etc methods).

Can we combine them in one class may be OR extract the common code?

Reply:
extracting common code to the base class to keep the code separate for leader and follower